map2loop/thickness_calculator.py

```diff
@@ -143,6 +156,190 @@ def compute(
         # For any unit thickness that still hasn't been calculated (i.e. at -1) set to
         # the mean thickness of the other units
         thicknesses["thickness"] = thicknesses.apply(
-            lambda row: mean_thickness if row["thickness"] == -1 else row["thickness"], axis=1
+            lambda row: mean_thickness if row["thickness"] == -1 else row["thickness"],
+            axis=1,
+        )
+        return thicknesses
+
+
+class ThicknessCalculatorBeta(ThicknessCalculator):
+    """
+    This class is a subclass of the ThicknessCalculator abstract base class. It implements the thickness calculation
+    method for a given set of interpolated data points.
+
+    Attributes:
+        thickness_calculator_label (str): A string that stores the label of the thickness calculator.
+        For this class, it is "ThicknessCalculatorBeta".
+
+    Methods:
+        compute(units: pandas.DataFrame, stratigraphic_order: list, basal_contacts: pandas.DataFrame, map_data: MapData)
+        -> pandas.DataFrame: Calculates a thickness map for the overall map area.
+    """
+
+    def __init__(self):
+        """
+        Initialiser for beta version of the thickness calculator
+        """
+        self.thickness_calculator_label = "ThicknessCalculatorBeta"
+        self.lines = []
+
+
+    @beartype.beartype
+    def compute(
+        self,
+        units: pandas.DataFrame,
+        stratigraphic_order: list,
+        basal_contacts: pandas.DataFrame,
+        structure_data: pandas.DataFrame,
+        map_data: MapData,
+    ) -> pandas.DataFrame:
+        """
+        Execute thickness calculator method takes unit data, basal_contacts, stratigraphic order, orientation data and
+        DTM data to estimate unit thickness.
+
+        The method works by iterating over the stratigraphic order of units. For each unit, it finds the basal contact
+        points and the top contact line. It then calculates the shortest line between these points. For each basal
+        contact point, it finds the interpolated points that are within 10% of the length of the shortest line. It then
+        calculates the true thickness of the unit at these points using the formula t = L . sin dip, where L is the
+        length of the line orthogonal to both contacts and dip is the dip of the interpolated points.
+        The method then calculates the median thickness and standard deviation for the unit.
+
+        Args:
+            units (pandas.DataFrame): the data frame of units to add thicknesses to
+            stratigraphic_order (list): a list of unit names sorted from youngest to oldest
+            basal_contacts (geopandas.GeoDataFrame): basal contact geo data with locations and unit names of the contacts (columns must contain ["ID","basal_unit","type","geometry"])
+            structure_data (pandas.DataFrame): sampled structural data
```

Comment:
as column names are specified for the basal contacts, in lines 210-211, it might be worthwhile to have the column names specified for structure data as well.

Reply:
As explained in #47, strucure_data will be removed in the near future.